### DISCUSSION TITLE
CommonJS support?

### DISCUSSION THREAD

**Discussion by jchimene** [rscarson/rustyscript#283] [Q&A] [answered]
After a few days, I think I'm getting closer to running a CommonJS script in Deno & Node. Running via rustyscript yields
``` Error: ReferenceError: require is not defined```
I'm loading the script based on the ```node_experimental``` example with a ```.cjs``` file type. 
I've also tried ``` "type": "commonjs" ``` in ``` package.json ```
with file type ``` .js ```

Can I expect CommonJS support via this crate? Or is it too soon?

The example @ [embedding-typescript-in-your-rust-project](https://cheatcod3.hashnode.dev/embedding-typescript-in-your-rust-project) hasen't aged well.

If it's too soon for CommonJS, I'll try to make that example work.

Thanks for your support!

===== Script =====
```
const scxml = require('@scion-scxml/scxml');
scxml.pathToModel("HealthChallenge.scxml", function (err, model) {
    if (err)
        throw err;
    model.prepare(function (err, fnModel) {
        if (err)
            throw err;
        //instantiate the interpreter
        const sc = new scxml.core.Statechart(fnModel);
        //start the interpreter
        sc.start();
        //send the init event
        console.log("RAD!");
        sc.gen({ name: "register", data: JSON.stringify({ "id": 10, "who": "Fred", "sms": "+15205551212" }) });
    });
});
```
Here's a sample run
```
$ deno   --allow-env --allow-read healthchallenge.cjs
RAD!
DEBUG [
  {
    name: "register",
    data: '{"id":10,"who":"Fred","sms":"+15205551212"}'
  },
  { id: 10, who: "Fred", sms: "+15205551212" },
  10
]
```
These ``` imports ``` "work" when the file type is ``` .ts ```
The ``` babel ``` loads; the ``` npm: ``` passes the "not a relative path" test, but descends into version hell
```
import * as babel from "babel-polyfill";
import * as scion from "npm:@scion-scxml/scxml";
```
Which leads into version hell
```
Error: ReferenceError: process is not defined
```

**Comment by jchimene** [comment]
I'm going switch to node from deno.

**Comment by rscarson** [comment, marked as answer]
While CJS _is_ suppported, it is only supported from inside npm modules, for the moment
